src/datawave_cli/base_interactions.py

```diff
@@ -0,0 +1,40 @@
+from abc import ABC, abstractmethod
+import sys
+
+
+class BaseInteractions(ABC):
+    def __init__(self, args):
+        self.namespace = args.namespace
+        self.init_base_url(args)
+        self.init_cert(args)
+
+    def init_cert(self, args):
+        if args.key is None:
+            self.cert = args.cert
+        else:
+            self.cert = (args.cert, args.key)
+
+    def init_base_url(self, args):
+        """
+        Define the base_url for interacting with datawave accumulo based on arguments
+        passed in at call. Has the possibilities of localhost:port, IP:port, or an actual
+        url.
+        """
+        if (args.localhost):
+            self.base_url = "https://localhost:8443"
+        elif (args.ip):
+            pod_ip = self.get_pod_ip()
+            self.base_url = f"https://{pod_ip}:8443"
+        else:
+            # TODO figure out a good default for the url

```

Comment:
with the catch on line 31, do we need a default now?

Reply:
args.url is pulling from that environment variable and defaults to `''` if not found. So I don't think we need a default value here.